bucket/poetry.json

```diff
@@ -9,10 +9,10 @@
     "installer": {
         "script": [
             "$env:POETRY_HOME=\"$dir\"",
-            "python3 \"$dir\\$fname\" --version $version"
+            "Invoke-Expression \"$(scoop which python) `\"$dir\\$fname`\" --version $version\""
```

Comment:
Unless you make the change mentioned in https://github.com/ScoopInstaller/Main/issues/4064#issuecomment-1292765454, using `scoop which python` is not going to make a difference - it will also point to the MStore one. 

So we don't need to use `scoop which python` here.

Reply:
In `poetry.json`, we use the `depends` to specify the scoop to install the python of scoop.

However, due to the Microsoft Store's `Path` policy, our `Python3` commands will be overwritten.
So using `scoop which python` will effectively find the address of the python installed by scoop rather than jumping to the Microsoft Store

Reply:
`scoop which python` picks up the first python.exe it finds in PATH - it does not care whether it was installed via Scoop or via MStore. So if the Scoop shim Path is preceded by MStore shim Path, it will return the MStore shim. 

This is why you need to read what the linked comment says, again.

Reply:
But in my computer, if I use `scoop which python`, it will return the true path for my python installed via scoop, but not return the MS store Path, so it can correctly identify the path of Python to install Poetry normally.
![image](https://user-images.githubusercontent.com/46255829/230827009-60062d1d-bbd7-44e9-8d92-7e258ffe5ae7.png)
![image](https://user-images.githubusercontent.com/46255829/230827568-9bed74c6-ae2c-4704-9479-906c78440c05.png)

So, the `scoop which python` can help us find the python installed by scoop without change the `Path` variable orders.
Also, with the `depends: python`, it can solve the python path rightly to install the poetry.

Reply:
What is the output of `$env:Path.Split(';')` on your system?

Reply:
I'm sure that if I go back to the original `python3` approach, I'll be redirected to MS store.
![image](https://user-images.githubusercontent.com/46255829/230892349-28e83ed7-e3ed-4e36-a01e-f62db17566ed.png)

Reply:
I find that, there are two similar `Path` for MS store.
![image](https://user-images.githubusercontent.com/46255829/230893901-708490b8-9fbe-43c4-bbdf-ec3a6a01a653.png)

I'm not sure if this is a unique case for me, but it' s really weird.🤣🤣